modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx

```diff
@@ -531,29 +529,92 @@ export function useObjectFieldForm({
 function AggregationSourceProperty({
 	disabled,
 	errors,
-	objectRelationships,
+	editingField,
+	objectDefinitionId,
 	objectFieldSettings = [],
 	setValues,
 }: IAggregationSourcePropertyProps) {
 	const [query, setQuery] = useState<string>('');
-	const [selectedRelatedObject, setSelectRelatedObject] = useState<
-		TObjectRelationship
-	>();
+	const [
+		selectedRelatedObjectDefinition,
+		setSelectRelatedObjectDefinition,
+	] = useState<TObjectRelationship>();
 	const [selectedSummarizeField, setSelectedSummarizeField] = useState<
 		string
 	>();
 	const [
 		selectedAggregationFunction,
 		setSelectedAggregationFunction,
 	] = useState<{label: string; value: string}>();
+	const [objectRelationships, setObjectRelatonships] = useState<
+		TObjectRelationship[]
+	>();
 	const [objectRelationshipFields, setObjectRelationshipFields] = useState<
 		ObjectField[]
 	>();
 
-	const handleChangeRelatedObject = async (
+	useEffect(() => {
+		const makeFetch = async () => {
+			setObjectRelatonships(
+				await fetchObjectRelationships(objectDefinitionId)
+			);
+		};
+
+		makeFetch();
+	}, [objectDefinitionId]);
+
+	useEffect(() => {
+		if (editingField && objectRelationships) {
+			const makeFetch = async () => {
+				const settings = normalizeFieldSettings(objectFieldSettings);
+
+				const [
+					currentRelatedObjectDefinition,
+				] = objectRelationships.filter(
```

Comment:
@danvb112 This could be a `find`, since you're only getting the first element of the returned array. Using `find` will avoid reading the rest of the array once the element desired is identified.